### PR TITLE
Fixing tests

### PR DESCRIPTION
**Pull request by kuba--**

### CODE REVIEW

**Comment by rbehjati** [comment]
Thanks for putting this together. I created #7 to add the other test file that we fixed together here.

**Comment by kuba--** [comment]
Ok, tests should compile and pass. I tried following on `gcc:13.2-bookworm` container:
```sh
cmake -DSPM_BUILD_TEST=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo -D SPM_ENABLE_TCMALLOC=on -DSPM_ENABLE_NFKC_COMPILE=on -DSPM_ENABLE_SHARED=off -DSPM_SANITIZE=off -DSPM_DISABLE_LTO=off -S . -B build -GNinja 

cmake --build build

cmake --build build --target test
[0/1] Running tests...
Test project /sentencepiece/build
    Start 1: sentencepiece_test
1/1 Test #1: sentencepiece_test ...............   Passed    0.01 sec

100% tests passed, 0 tests failed out of 1
```

File: src/bpe_model_test.cc [hunk 1]
```diff
@@ -69,7 +69,9 @@ TEST(BPEModelTest, EncodeTest) {
   model_proto.mutable_pieces(12)->set_type(  // r
       ModelProto::SentencePiece::USER_DEFINED);
 
-  const Model model(model_proto);
+
+  std::unique_ptr<const ModelProto> mp(new ModelProto(model_proto));
```
Comment (vmarkovtsev) [inline src/bpe_model_test.cc:73]: pls use `std::make_unique`:

```c++
auto mp = std::make_unique<const ModelProto>(model_proto);
```

or even better

```c++
const Model model(std::make_unique<const ModelProto>(model_proto));
```

**Review by vmarkovtsev — CHANGES_REQUESTED** [review]

File: src/bpe_model_test.cc (hunk 1 above)
Comment (kuba--) [reply src/bpe_model_test.cc]: All changes are refined accordingly

**Review by vmarkovtsev — APPROVED** [review]

**Pull request merged**